state_test.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2018 Tuenti Technologies S.L. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pouch
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"reflect"
+	"sort"
+	"testing"
+)
+
+var filesUsingCases = []struct {
+	Secret          *SecretState
+	RegisteredFiles *FileSortedList
+	SortedFiles     *FileSortedList
+}{
+	{
+		secretWithFilesUsing,
+		filesUsingPriorities,
+		sortedFilesUsingPriorities,
+	},
+	{
+		secretWithFilesUsing,
+		filesUsingNoPriorities,
+		sortedFilesUsingNoPriorities,
+	},
+}
+
+var filesUsingPriorities = &FileSortedList{
+	PriorityFile{Priority: 10, Path: "/tmp2"},
+	PriorityFile{Priority: 10, Path: "/tmp1"},
+	PriorityFile{Priority: 90, Path: "/tmp3a"},
+	PriorityFile{Path: "/bar"},
+	PriorityFile{Priority: 20, Path: "/tmp3b"},
+}
+
+var sortedFilesUsingPriorities = &FileSortedList{
+	PriorityFile{Path: "/bar"},
+	PriorityFile{Priority: 10, Path: "/tmp1"},
+	PriorityFile{Priority: 10, Path: "/tmp2"},
+	PriorityFile{Priority: 20, Path: "/tmp3b"},
+	PriorityFile{Priority: 90, Path: "/tmp3a"},
+}
+
+var secretWithFilesUsing = &SecretState{
+	Data:       map[string]interface{}{"test_key": "value_secret"},
+	FilesUsing: FileSortedList{},
+}
+
+var filesUsingNoPriorities = &FileSortedList{
+	PriorityFile{Path: "/temp2"},
+	PriorityFile{Path: "/temp1"},
+	PriorityFile{Path: "/tempcc"},
+	PriorityFile{Path: "/tempbb"},
+	PriorityFile{Path: "/other"},
+}
+
+var sortedFilesUsingNoPriorities = &FileSortedList{
+	PriorityFile{Path: "/other"},
+	PriorityFile{Path: "/temp1"},
+	PriorityFile{Path: "/temp2"},
+	PriorityFile{Path: "/tempbb"},
+	PriorityFile{Path: "/tempcc"},
+}
+
+func TestSortingFilesUsing(t *testing.T) {
+
+	var filesUsing FileSortedList
+	for _, c := range filesUsingCases {
+		filesUsing = *c.RegisteredFiles
+		sort.Sort(filesUsing)
+
+		if !reflect.DeepEqual(filesUsing, *c.SortedFiles) {
+			t.Fatalf("Sorting PriorityFiles errors %s %s", filesUsing, *c.SortedFiles)
```

Comment:
Sorting errors? it is just not ordered as expected :)

Reply:
True, re-stated that message